src/CalcViewModel/Common/NavCategory.cpp

```diff
@@ -48,45 +47,57 @@ static constexpr int CURRENCY_ID = 16;
 static constexpr int GRAPHING_ID = 17;
 // ^^^ THESE CONSTANTS SHOULD NEVER CHANGE ^^^
 
-wchar_t* towchar_t(int number)
+namespace
 {
-    auto wstr = to_wstring(number);
-    return _wcsdup(wstr.c_str());
-}
-
-bool IsGraphingModeAvailable()
-{
-    static bool supportGraph = Windows::Foundation::Metadata::ApiInformation::IsMethodPresent("Windows.UI.Text.RichEditTextDocument", "GetMath");
-    return supportGraph;
-}
+    Platform::Agile<Windows::System::User^> CurrentUser;
+    std::mutex GraphingModeCheckMutex;
 
-Box<bool> ^ _isGraphingModeEnabledCached = nullptr;
-bool IsGraphingModeEnabled(User ^ currentUser = nullptr)
-{
-    if (!IsGraphingModeAvailable())
+    wchar_t* towchar_t(int number)
     {
-        return false;
+        auto wstr = std::to_wstring(number);
+        return _wcsdup(wstr.c_str());
     }
 
-    if (_isGraphingModeEnabledCached != nullptr)
+    bool IsGraphingModeAvailable()
     {
-        return _isGraphingModeEnabledCached->Value;
+        static bool supportGraph = Windows::Foundation::Metadata::ApiInformation::IsMethodPresent("Windows.UI.Text.RichEditTextDocument", "GetMath");
```

Comment:
GetMath is available in Windows 10 version 19041, and these days Calculator does not support OSes older than 19041. So I think we can just remove this check altogether. Graphing mode is always "available".

You might want to submit a separate change to remove this check, since I think it might have a measurable performance improvement even without the rest of this refactoring.

Reply:
Nice suggestion.
I removed the check and adjusted the parts that depend on this precondition.